ddmrp/models/stock_warehouse_orderpoint.py

```diff
@@ -226,6 +229,90 @@ def _compute_ddmrp_chart(self):
             script, div = components(p)
             rec.ddmrp_chart = '%s%s' % (div, script)
 
+    def _compute_ddmrp_demand_supply_chart(self):
+        for rec in self:
+            if not rec.buffer_profile_id:
+                # Not a buffer, skip.
+                rec.ddmrp_demand_chart = ''
+                rec.ddmrp_supply_chart = ''
+                continue
+
+            # Prepare data:
+            demand_data = rec._get_demand_by_days()
+            supply_data = rec._get_incoming_by_days()
+            width = timedelta(days=0.4)
+
+            # Plot demand data:
+            if demand_data:
+                x_demand = list(demand_data.keys())
+                y_demand = list(demand_data.values())
+
+                p = figure(plot_width=500, plot_height=400,
+                           y_axis_label='Quantity', x_axis_type='datetime')
+                p.toolbar.logo = None
+                p.sizing_mode = 'stretch_both'
+                # TODO: # p.xaxis.label_text_font = 'helvetica'
+                p.xaxis.formatter = DatetimeTickFormatter(
+                    hours=["%d %B %Y"], days=["%d %B %Y"], months=["%d %B %Y"],
+                    years=["%d %B %Y"])
+                p.xaxis.major_label_orientation = pi / 4
+
+                p.vbar(x=x_demand, width=width, bottom=0, top=y_demand,
+                       color="firebrick")
+                p.line(
+                    [datetime.today() - timedelta(days=1),
+                     datetime.today() + timedelta(days=rec.order_spike_horizon)],
+                    [rec.order_spike_threshold, rec.order_spike_threshold],
+                    line_width=2, line_dash='dashed')
+
+                unit = rec.product_uom.name
+                hover = HoverTool(
+                    tooltips=[("qty", "$y %s" % unit)],
```

Comment:
You can use .format:
```suggestion
                    tooltips=[("qty", "$y {}".format(unit))],
```

Reply:
See [contributing guidelines](https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#33idioms):

> Prefer % over .format(), prefer %(varname) instead of positional. This is better for translation and clarity.

Reply:
Ok, but maybe we should change that in the guidelines:
https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations

@moylop260 What do you think?

Reply:
Sorry, We tested and using format continues to cause problems with translations.
Forget my comment.